smithy-aws-cloudformation/src/test/java/software/amazon/smithy/aws/cloudformation/schema/fromsmithy/ResourceSchemaTest.java

```diff
@@ -0,0 +1,61 @@
+/*
+ * Copyright 2020 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+package software.amazon.smithy.aws.cloudformation.schema.fromsmithy;
+
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.MethodSource;
+import software.amazon.smithy.aws.cloudformation.schema.model.ResourceSchema;
+import software.amazon.smithy.model.node.Node;
+import software.amazon.smithy.model.node.NodeMapper;
+import software.amazon.smithy.utils.IoUtils;
+
+import java.io.IOException;
+import java.net.URISyntaxException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.List;
+import java.util.stream.Collectors;
+
+public class ResourceSchemaTest {
+
+    @ParameterizedTest
+    @MethodSource("resourceSchemaFiles")
+    public void validateResourceSchemaFromNodeToNode(String resourceSchemaFile) {
+        NodeMapper mapper = new NodeMapper();
+        String json = IoUtils.readUtf8File(resourceSchemaFile);
+
+        Node node = Node.parse(json);
+        ResourceSchema schemaFromNode = mapper.deserialize(node, ResourceSchema.class);
+        Node nodeFromSchema = schemaFromNode.toNode();
+
+        Node.assertEquals(nodeFromSchema.withDeepSortedKeys(), node.withDeepSortedKeys());
+    }
+
+    public static List<String> resourceSchemaFiles() {
+        try {
+            Path definitionPath = Paths.get(ResourceSchemaTest.class.getResource("aws-sagemaker-domain.json").toURI());
```

Comment:
Should this be `.cfn.` file to be included in the test suite?

Reply:
Interestingly, adding that to the filename causes the node comparison to break - the order of array items in the node change... 

I think the way we deep-sort is incorrect.

Reply:
It could be that or something isn't using an ordered collection.

Reply:
Ah, yep.

Reply:
Took a crack at it and defined a custom sorter, but decided against it. I just updated the cfn file to have correctly ordered items.